frontend/src/app/shared/interceptors/error.interceptor.ts

```diff
@@ -0,0 +1,44 @@
+import { HttpEvent, HttpHandler, HttpInterceptor, HttpRequest } from '@angular/common/http';
+import { Observable, catchError, switchMap } from 'rxjs';
+
+import { AuthService } from '@api/auth.service';
+import { Injectable } from '@angular/core';
+import { Router } from '@angular/router';
+import { Store } from '@ngrx/store';
+import { addToken } from '@store/actions/login-page.actions';
+import { selectToken } from '@store/selectors/auth-page.selectors';
+
+@Injectable()
+export class ErrorInterceptor implements HttpInterceptor {
+    constructor(private authService: AuthService, private store: Store, private router: Router) {}
+    intercept(req: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
+        return next.handle(req).pipe(
```

Comment:
1. How do you handle failed refresh token request? Would user be logged out? See any messages? 
2. What if there would be many 401 resonses? it would send many refresh token requests?
3. it would be correct to collect all requests with 401 response and after token request send them again with new token. From user prespective it will looks like just longer loading.